core/frontend/apps/amp/lib/helpers/amp_components.js

```diff
@@ -23,7 +23,14 @@ function ampComponents() {
         components.push('<script async custom-element="amp-anim" src="https://cdn.ampproject.org/v0/amp-anim-0.1.js"></script>');
     }
 
-    if (html.indexOf('<iframe') !== -1) {
+    let iframeCount = (html.match(/<iframe/g) || []).length,
+        youtubeCount = (html.match(/youtube.com\/embed/g) || []).length;
```

Comment:
This regex won't catch all YouTube embeds, we should probably use a similar regex to the [one used in the amp conversion library](https://github.com/jbhannah/amperize/blob/master/lib/amperize.js#L299).

`/(youtu.be\/|youtube(-nocookie)?.com\/(v\/|.*u\/\w\/|embed\/|.*v=))/` might be a reasonable starting point?

Reply:
I will take a look and fix that!

Reply:
@kevinansfield Your suggestion works and I've added it. This also works, but i'm not sure getting the video ID really helps.
`/(youtu.be\/|youtube(-nocookie)?.com\/(v\/|.*u\/\w\/|embed\/|.*v=))([\w-]{11}).*/g`

This is all still vulnerable to mistaking text as a youtube embedded video. For example, `<p>youtube.com/embed/</p>` will cause the amp-youtube script to be added. I can add a test for that if you'd like.